frontend/public/components/utils/kebab.tsx

```diff
@@ -103,34 +103,69 @@ export const ResourceKebab = connectToModel((props: ResourceKebabProps) => {
   />;
 });
 
-export class Kebab extends DropdownMixin {
+export class Kebab extends React.Component<KebabProps, KebabState> {
```

Comment:
If we're rewriting this, we should go ahead and use React hooks. See #1456

Reply:
hey @spadgett - React Hooks render like stateless functions it seems, and we are currently exposing `kebabFactory` as a static member to other React components:
```
export class Kebab extends React.Component<KebabProps, KebabState> {
  static factory: KebabFactory = kebabFactory;
```
consumed like this in other places (such as [build.tsx](https://github.com/openshift/console/blob/master/frontend/public/components/build.tsx#L22):
```
const { common, EditEnvironment } = Kebab.factory;
```
Do you want to expose these factory constants on a different class and still convert Kebab to a Hook function?

Just some clarification first... there is several references to `Kebab.factory` throughout

Reply:
Rewriting the inner function as `KebabHook` would seem to have the same effect without changing the global if this works for you...
https://github.com/priley86/web-ui/commit/894a8ca155c8ab9bc84ef0bcb934fa5b8b1fffee

Reply:
Yeah, I don't see a reason it needs to be a static member. I'd just pull it outside of the class. We're trying to use functional components instead of classes when possible.

Reply:
ok - happy to update w/ this, but will wait until your other kebab refactor for RBAC finishes

Reply:
rebased RBAC changes, added Kebab Hook... seems to work well here. I will need to add the additional custom kebab CSS later alongside this PR. Resolving for now.